diff --git a/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c b/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c
index 711e6fe..519ea04 100644
--- a/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c
+++ b/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c
@@ -290,8 +290,8 @@ hm_atomic_insert(PMEMobjpool *pop, TOID(struct hashmap_atomic) hashmap,
 
 
 	PM_EQU(D_RW(hashmap)->count_dirty, 0);
-	pmemobj_persist(pop, &D_RW(hashmap)->count_dirty,
-			sizeof(D_RW(hashmap)->count_dirty));
+	// pmemobj_persist(pop, &D_RW(hashmap)->count_dirty,
+	// 		sizeof(D_RW(hashmap)->count_dirty));
 
 	PMTest_isPersistedBefore(&D_RW(hashmap)->count, sizeof(D_RW(hashmap)->count), &D_RW(hashmap)->count_dirty, sizeof(D_RW(hashmap)->count_dirty));
 
