diff --git a/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c b/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c
index 711e6fe..b03f093 100644
--- a/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c
+++ b/nvml/src/examples/libpmemobj/hashmap/hashmap_atomic.c
@@ -332,8 +332,6 @@ hm_atomic_remove(PMEMobjpool *pop, TOID(struct hashmap_atomic) hashmap,
 		return OID_NULL;
 
 	PM_EQU(D_RW(hashmap)->count_dirty, 1);
-	pmemobj_persist(pop, &D_RW(hashmap)->count_dirty,
-			sizeof(D_RW(hashmap)->count_dirty));
 
 	if (POBJ_LIST_REMOVE_FREE(pop, &D_RW(buckets)->bucket[h],
 			var, list)) {
@@ -344,6 +342,8 @@ hm_atomic_remove(PMEMobjpool *pop, TOID(struct hashmap_atomic) hashmap,
 
 
 	PM_EQU(D_RW(hashmap)->count, D_RO(hashmap)->count - 1);
+	pmemobj_persist(pop, &D_RW(hashmap)->count_dirty,
+			sizeof(D_RW(hashmap)->count_dirty));
 	pmemobj_persist(pop, &D_RW(hashmap)->count,
 			sizeof(D_RW(hashmap)->count));
 
