diff --git a/nvml/src/examples/libpmemobj/map/data_store.c b/nvml/src/examples/libpmemobj/map/data_store.c
index 075bd88..67ab9fa 100755
--- a/nvml/src/examples/libpmemobj/map/data_store.c
+++ b/nvml/src/examples/libpmemobj/map/data_store.c
@@ -253,8 +253,11 @@ int main(int argc, const char *argv[]) {
 		PMTest_START;
 		if (strcmp("hashmap_atomic", type))
 			PMTest_CHECKER_START;
-
-		PMEMoid item = map_remove(mapc, D_RW(root)->map, keys[i]);
+		PMEMoid item;
+		if (i % 2 == 0)
+			item = map_remove(mapc, D_RW(root)->map, keys[i/2]);
+		else
+			item = map_remove(mapc, D_RW(root)->map, keys[nkeys-(i+1)/2]);
 	
 		if (strcmp("hashmap_atomic", type))
 			PMTest_CHECKER_END;
