diff --git a/nvml/src/examples/libpmemobj/hashmap/hashmap_tx.c b/nvml/src/examples/libpmemobj/hashmap/hashmap_tx.c
index 637369d..2d5426c 100644
--- a/nvml/src/examples/libpmemobj/hashmap/hashmap_tx.c
+++ b/nvml/src/examples/libpmemobj/hashmap/hashmap_tx.c
@@ -201,6 +201,7 @@ hm_tx_insert(PMEMobjpool *pop, TOID(struct hashmap_tx) hashmap,
 		TX_ADD_FIELD(D_RO(hashmap)->buckets, bucket[h]);
 		//TX_ADD_FIELD(buckets, bucket[h]);
 		TX_ADD_FIELD(hashmap, count);
+		TX_ADD_FIELD(hashmap, count);
 		
 		TOID(struct entry) e = TX_NEW(struct entry);
 		PMTest_assign(&(D_RW(e)->key), sizeof(struct entry));
