diff --git a/nvml/src/examples/libpmemobj/tree_map/btree_map.c b/nvml/src/examples/libpmemobj/tree_map/btree_map.c
index 57c3b35..3b0890a 100644
--- a/nvml/src/examples/libpmemobj/tree_map/btree_map.c
+++ b/nvml/src/examples/libpmemobj/tree_map/btree_map.c
@@ -322,7 +322,7 @@ btree_map_rotate_right(TOID(struct tree_map_node) rsb,
 	btree_map_insert_item(node, D_RO(node)->n, sep);
 
 	/* the first element of the right sibling is the new separator */
-	TX_ADD_FIELD(parent, items[p]);
+	// TX_ADD_FIELD(parent, items[p]);
 	PM_EQU(D_RW(parent)->items[p], D_RO(rsb)->items[0]);
 
 	/* the nodes are not necessarily leafs, so copy also the slot */
