diff --git a/nvml/src/examples/libpmemobj/tree_map/btree_map.c b/nvml/src/examples/libpmemobj/tree_map/btree_map.c
index 57c3b35..1aaed67 100644
--- a/nvml/src/examples/libpmemobj/tree_map/btree_map.c
+++ b/nvml/src/examples/libpmemobj/tree_map/btree_map.c
@@ -392,7 +392,7 @@ btree_map_merge(TOID(struct btree_map) map, TOID(struct tree_map_node) rn,
 
 	TX_FREE(rn); /* right node is now empty */
 
-	TX_ADD(parent);
+	// TX_ADD(parent);
 	PM_EQU(D_RW(parent)->n, D_RO(parent)->n - 1);
 
 	/* move everything to the right of the separator by one array slot */
